app/code/Magento/CatalogRule/Controller/Adminhtml/Promo/Widget/CategoriesJson.php

```diff
@@ -32,7 +33,7 @@ public function __construct(Context $context, Registry $coreRegistry)
     /**
      * Initialize category object in registry
      *
-     * @return Category
+     * @return Category|Boolean
```

Comment:
How this method can return `Boolean` class?

Reply:
@orlangur
Method returns false value at if $rootid not found in category path ids.

`if (!in_array($rootId, $category->getPathIds())) {`
 `       $this->_redirect('catalog/*/', ['_current' => true, 'id' => null]);`
`        return false;`
`   }`

Reply:
`false` is `bool` primitive type, it has nothing to do with `Boolean` class.

Reply:
Okay.Thanks for helping. Should i add simple `boolean` or leave it as it is and remove file from PR?

Reply:
Add `|bool` and remove unneeded class import.

Reply:
I update the file. Please review it.